packages/wonder-blocks-tooltip/util/visibility-modifier.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+/**
+ * A modifier for use with popper.js or react-popper.
+ * This looks at the scroll parents of the reference element to determine
+ * overall visibility.
+ */
+import {default as PopperJS} from "popper.js";
+
+import {getElementIntersection} from "@khanacademy/wonder-blocks-core";
+
+import isObscured from "./is-obscured.js";
+
+/**
+ * The function that implements the modifier.
+ */
+function visibilityModifierFn(data: any) {
+    const anchorElement = data.instance.reference;
+
+    // First, we see how the element intersects with its scroll parents.
+    // If it doesn't, then we should hide it.
+    // Otherwise, we check to see if anything else obscures the component (like
+    // a fixed or absolute positioned element).
+    const {horizontal, vertical} = getElementIntersection(anchorElement);
+    const hide =
+        horizontal !== "within" ||
+        vertical !== "within" ||
+        isObscured(anchorElement);
+
+    // If we're hidden, we mimic what the built-in hide method does,
+    // and set the hide flag and the OOB attribute with appropriate
+    // short-circuiting.
+    if (hide) {
+        if (data.hide) {
+            return data;
+        }
+        data.hide = true;
+        data.attributes["x-out-of-boundaries"] = "";
```

Comment:
Is `"x-out-of-boundaries"` only used for test purposes?

Reply:
I've added a link in the code. Reproduced here:

https://github.com/FezVrasta/popper.js/blob/08c5d6010346bf9df06e9f81a54fa6c2c51e3639/packages/popper/src/modifiers/hide.js#L29-L42